packages/ethereum-contracts/dev-scripts/deploy-test-framework.js

```diff
@@ -65,21 +76,36 @@ const _getFactoryAndReturnDeployedContract = async (
         signerOrOptions
     );
     const contract = await ContractFactory.deploy(...args);
-    await contract.deployed();
+
+    // ethers v5
+    if (contract.deployed) {
+        await contract.deployed();
+    } else if (!contract.deployed) {
+        // ethers v6
+        await contract.waitForDeployment();
+    }
+
     if (process.env.DEBUG_CONSOLE === true) {
-        console.log(`${contractName} Deployed At:`, contract.address);
+        console.log(
+            `${contractName} Deployed At:`,
+            getContractAddress(contract)
+        );
     }
     return contract;
 };
 
 /**
  * Deploys Superfluid Framework in local testing environments.
- * NOTE: This only works with Hardhat.
+ * NOTE: This only works with Hardhat + ethers v5/ethers v6 currently.
+ * @param privateKey NEVER USE A PRIVATE KEY WITH REAL FUNDS - a test account private key
+ * @param provider an ethers provider
+ * @param ethersV5Signer an ethers v5 signer
  * @returns
  */
-const deployTestFramework = async () => {
-    const signer = (await ethers.getSigners())[0];
-    await deployERC1820(ethers.provider);
+const deployTestFramework = async (privateKey, provider, ethersV5Signer) => {
```

Comment:
I think this interface is a bit confusing. Can we change it such that it better conveys that either privateKey or ethersV5Signer is needed?

Reply:
one of the downsides of javascript :/

do you have something in mind?